plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundPartition.java

```diff
@@ -38,101 +36,73 @@
+        } catch (GroundException e) {
+            LOG.error("Unable to fetch parition node structure");
+            throw new GroundException(e);
+        }
     }
-  }
 
-  public Structure getEdgeStructure(String partitionName) throws GroundException {
-    try {
-      Edge edge = getEdge(partitionName);
-      return groundReadWrite.getStructureFactory().retrieveFromDatabase(partitionName);
-    } catch (GroundException e) {
-      LOG.error("Unable to fetch table partition edge structure");
-      throw e;
+    public Edge getEdge(String partitionName) throws GroundException {
+        LOG.debug("Fetching table partition edge: " + partitionName);
+        return groundReadWrite.getGroundReadWriteEdgeResource().getEdge(partitionName);
     }
-  }
-
-  public NodeVersion createPartition(String dbName, String tableName, Partition part)
-      throws InvalidObjectException, MetaException {
-    try {
-      ObjectPair<String, String> objectPair = new ObjectPair<>(HiveStringUtils.normalizeIdentifier(dbName),
-          HiveStringUtils.normalizeIdentifier(tableName));
-      String partId = objectPair.toString();
-      for (String value : part.getValues()) {
-        partId += ":" + value;
-      }
 
-      Tag partTag = new Tag(0, partId, JsonUtil.toJSON(part), GroundType.STRING);
-
-      Node node = this.getNode(partId);
-      long nodeId = node.getId();
-      Structure partStruct = this.getNodeStructure(partId);
-      Map<String, GroundType> structVersionAttribs = new HashMap<>();
-      structVersionAttribs.put(partId, GroundType.STRING);
-      StructureVersion sv = groundReadWrite.getStructureVersionFactory().create(partStruct.getId(), structVersionAttribs,
-          new ArrayList<Long>());
-
-      String reference = part.getSd().getLocation();
-      HashMap<String, Tag> tags = new HashMap<>();
-      tags.put(partId, partTag);
-
-      long versionId = sv.getId();
-      List<Long> parentId = new ArrayList<>();
-
-      Map<String, String> parameters = part.getParameters();
+    public Structure getEdgeStructure(String partitionName) throws GroundException {
+        try {
+            Edge edge = getEdge(partitionName);
+            return groundReadWrite.getGroundReadWriteStructureResource().getStructure(edge.getName());
+        } catch (GroundException e) {
+            LOG.error("Unable to fetch table partition edge structure");
+            throw new GroundException(e);
+        }
+    }
 
-      return groundReadWrite.getNodeVersionFactory().create(tags, versionId, reference, parameters,
-          nodeId, parentId);
-    } catch (GroundException e) {
-      throw new MetaException("Unable to create partition " + e.getMessage());
+    public NodeVersion createPartition(String dbName, String tableName, Partition part)
+            throws InvalidObjectException, MetaException {
+        try {
+            ObjectPair<String, String> objectPair = new ObjectPair<>(HiveStringUtils.normalizeIdentifier(dbName),
```

Comment:
am not sure about spacing.Use google style code formatter through out. can you be specific?